.mailmap

```diff
@@ -48,6 +52,7 @@ Bjoern A. Zeeb <baz21@cam.ac.uk>
 Blake Hechtman <bah13@duke.edu> Blake Hechtman <blake.hechtman@amd.com>
 Blake Hechtman <bah13@duke.edu> Blake Hechtman ext:(%2C%20Nilay%20Vaish%20%3Cnilay%40cs.wisc.edu%3E) <bah13@duke.edu>
 Bobby R. Bruce <bbruce@ucdavis.edu> Bobby Bruce <bbruce@amarillo.cs.ucdavis.edu>
+Bobby R. Bruce <bbruce@ucdavis.edu> Bobby Bruce <bbruce@ucdavis.edu>
 Boris Shingarov <shingarov@gmail.com> Boris Shingarov <shingarov@labware.com>
 Brad Beckmann <brad.beckmann@amd.com> Brad Beckmann <Brad.Beckmann@amd.com>
 Brad Beckmann <brad.beckmann@amd.com> Brad Beckmann ext:(%2C%20Nilay%20Vaish%20%3Cnilay%40cs.wisc.edu%3E) <Brad.Beckmann@amd.com>
```

Comment:
Probably we want line 58 to be fixed as Nilay's email address is embedded in Brad's info?

Also I think 57 and 58 are identical except for this issue.

Reply:
I think there may be some confusion about the purpose/syntax of this file. Here's link to the documentation: https://git-scm.com/docs/gitmailmap

Basically, each line is `[correct info] [incorrect info that should be the info on the left]`.

So, in this case, Brad made a change, but Nilay committed on behalf of brad (because this was how it worked with mercurial before we moved to git). So, the commit was authored by Brad, not Nilay. Nilay was just the one that happened to push upstream (again, mercurial was different from the way git works and the conversion from hg to git wasn't perfect).

So, this is correct. Anytime there is a commit that says "Brad Beckmann ext:(..." we should interpret this as "Brad Beckmann <brad.beckmann@amd.com>".

The main point of this file is so we can track community stats. So, we want to make sure that we have unique (and preferably correct) names on the left hand side of each line. This will allow us to answer questions like "how many people have contributed to the gem5 codebase" or "what is the number of committers with more that 10 commits in the past 2 years."

Finally, I'll point out this *doesn't change the commit info at all*. So, all of the details of which institution is still available. I believe that `git shortlog` and a couple of other commands like that are the only times the mailmap file is used.

Reply:
ok, as long as the ones I pointed out that seem duplicated are intentional, then that's fine